### ISSUE TITLE
Model doesn't work when used without existingConnection

### ISSUE THREAD

**Issue by kussberg** [typegoose/typegoose#948]
Using latest version of typegoose and mongoose.

This code works:
```ts
const dbConnection = await connectToMongoDB();
const IssueModel = getModelForClass(IssueClass, {
      existingConnection: dbConnection,
});
```

This code doesn't:
```ts
await connectToMongoDB();
const IssueModel = getModelForClass(IssueClass, {
      existingConnection: dbConnection,
});
```

For some context this is `dbConnection` file:
```ts
import mongoose, { Connection } from "mongoose";

let cachedConnection: Connection | null = null;

export default async function connectToMongoDB() {
  if (cachedConnection) {
    console.log("Using cached db connection");
    return cachedConnection;
  }
  try {
    const cnx = await mongoose.connect(process.env.MONGO_URI!);
    cachedConnection = cnx.connection;
    console.log("New mongodb connection established");
    return cachedConnection;
  } catch (error) {
    console.log(error);
    throw error;
  }
}

```

Error that I am getting when I am not explicitly providing existing db connection:
`MongooseError: Operation `issues.find()` buffering timed out after 10000ms`

**Comment by ollyde** [comment]
I'd like to also expand on this, here's a working example that breaks without existing connection.

@modelOptions({
	schemaOptions: { collection: 'app_details' }
})
export class AppDetails extends TimeStamps {
	@prop({ required: true, type: String })
	public version!: string;

	@prop({ required: true, type: Number })
	public numberOfBoots!: number;
}

export const AppDetailsModel = getModelForClass(AppDetails, {
	existingConnection: mongoose.connection // Remove this and it fails on all getModelForClass objects.
});

/**
 * Connect to MongoDB using native driver and Mongoose
 */
export async function connectDB() {
	try {
		// First connect with native driver
		const client = new MongoClient(MONGO_URI, {
			serverSelectionTimeoutMS: 15000,
			socketTimeoutMS: 30000
		});

		await client.connect();
		const db = client.db(APP_NAME);

		// Test native connection
		await db.admin().ping();
		console.log('Native MongoDB connection successful');

		// Now connect Mongoose using the existing native connection
		await mongoose.connect(MONGO_URI, {
			dbName: APP_NAME,
			autoCreate: true,
			autoIndex: true
		});

		console.log('Mongoose connected using native connection');

		// Test database writes
		const appDetails = await AppDetailsModel.findOne();
		if (!appDetails) {
			await AppDetailsModel.create({
				version: '0.0.0',
				numberOfBoots: 1
			});
		} else {
			await AppDetailsModel.updateOne({ _id: appDetails._id }, { $inc: { numberOfBoots: 1 } });
		}

		console.log('Database writes successful');

		return { client, mongoose };
	} catch (error) {
		console.error('Database connection error:', error);
		throw error;
	}
}

**Comment by hasezoey** [comment]
@ollyde Cannot reproduce this with your example code, please verify you only have one mongoose version installed (`yarn why mongoose` / `npm ls mongoose`), otherwise please modify the repository below to reproduce this.

<details>
<summary>Output of the script</summary>

```txt
Native MongoDB connection successful
Mongoose connected using native connection
Database writes successful
````

</details>

[Reproduction Repository & branch](https://github.com/typegoose/typegoose-testing/tree/verify948)

**Comment by ollyde** [comment]
@hasezoey it works on production when I do

`import { mongoose as typegoose } from '@typegoose/typegoose';`

instead of mongoose.

```
0.0.1 /Users/starfire/Projects/myproject
└─┬ @typegoose/typegoose@12.11.0
  └── mongoose@8.10.0
```

**Comment by ollyde** [comment]
Yes I spent a long time figuring this out. Maybe typegoose should have it's own connection name/method that references the correct one?

**Comment by hasezoey** [comment]
> Maybe typegoose should have it's own connection name/method that references the correct one?

Typegoose does not handle connections, it is literally just a fancy way of generating a schema and then registering it on a mongoose instance / connection, which is why you can provide `existingConnection` and `existingMongoose`.

If there is duplication in the dependencies for whatever reason, i dont think mongoose nor typegoose can do much about it.
(as mongoose only knows about itself, and typegoose only knows about the mongoose that it has access to)

**Comment by ollyde** [comment]
@hasezoey I know, just would be a better way to ensure they use the right import ;-)

**Comment by hasezoey** [comment]
I will close this as it has been stale for over 30 days.
